File: 461.java
class Solution {
    public int hammingDistance(int x, int y) {
        // time: O(1), space: O(1)
        int hamming = 0;
        int xor = x ^ y;
        while (xor != 0) {
            if ((xor & 1) == 1) {
                hamming++;
            }
            xor >>= 1;
        }
        return hamming;
    }
}

File: 461.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        # time: O(1), space: O(1)
        hamming = 0
        while x and y:
            if (x & 1) + (y & 1) == 1:
                hamming += 1
            x >>= 1
            y >>= 1
        while x:
            if x & 1 == 1:
                hamming += 1
            x >>= 1
        while y:
            if y & 1 == 1:
                hamming += 1
            y >>= 1
        return hamming